src/expr.cpp

```diff
@@ -2039,6 +2039,22 @@ const Type *BinaryExpr::GetType() const {
         }                                                                                                              \
         break
 
+template <typename T> static T countLeadingZeros(T val) {
+
+    T leadingZeros = 0;
+    T size = sizeof(T) * CHAR_BIT;
```

Comment:
sizeof(T) is ``size_t``, so you will be downconverting in some cases when assigning to ``T`` type. Some compilers will have warnings. I suggest ``size`` to make ``size_t`` and explicitly convert ``1 << (size - 1)`` to ``T``.